### PR TITLE
Decomposed send_post_request method to abide the SRP principle

### PR DESCRIPTION
**Pull request by gokul-koganti**
Decomposed send_post_request method into encrypt_request_body, format_into_JSON, decrypt_response, update_participants, process_response_body, add_expert_grades, add_quiz_scores, add_hamer_reputation_values, add_lauw_reputation_values, prepare_request_body. The control upon entering send_post_request method will call prepare_request_method and gets the encrypted request body. The method then sends the request and passes the response to the process_response_body method to process and update the participants.

prepare_request_method uses add_expert_grades, add_quiz_scores, add_hamer_reputation_values, add_lauw_reputation_values, encrypt_request_body, format_into_JSON to prepare the request body.

process_response_body makes use of decrypt_response and update_participants to process the response received.

All the decomposed functions abide by the single responsibility principle.

### CODE REVIEW

File: app/controllers/reputation_web_service_controller.rb [hunk 1]
```diff
@@ -163,87 +250,32 @@ def send_post_request
...
-      when '754' # expert grades of Wiki contribution (754)
-        req.body.prepend('"expert_grades": {"submission25030":95,"submission25031":92,"submission25033":88,"submission25034":98,"submission25035":100,"submission25037":95,"submission25038":95,"submission25039":93,"submission25040":96,"submission25041":90,"submission25042":100,"submission25046":95,"submission25049":90,"submission25050":88,"submission25053":91,"submission25054":96,"submission25055":94,"submission25059":96,"submission25071":85,"submission25082":100,"submission25086":95,"submission25097":90,"submission25098":85,"submission25102":97,"submission25103":94,"submission25105":98,"submission25114":95,"submission25115":94},')
-      when '756' # expert grades of Wikipedia contribution (756)
-        req.body.prepend('"expert_grades": {"submission25107":76.6667,"submission25109":83.3333},')
-        # comment end
-      end
+      req.body = add_expert_grades(req.body)
```
Comment (krishnasaurabh) [inline app/controllers/reputation_web_service_controller.rb]: You are not returning anything in add_expert_grades, please make sure this works, by running a stub in the test/ or any way possible.

File: app/controllers/reputation_web_service_controller.rb [hunk 2]
```diff
@@ -163,87 +250,32 @@ def send_post_request
...
-      @@additional_info = 'add quiz scores'
-      assignment_id_list_quiz = get_assignment_id_list(params[:assignment_id].to_i, params[:another_assignment_id].to_i)
-      quiz_str =  generate_json_for_quiz_scores(assignment_id_list_quiz).to_json
-      # quiz_str = json_generator(params[:assignment_id].to_i, params[:another_assignment_id].to_i, params[:round_num].to_i, 'quiz scores').to_json
-      quiz_str[0] = ''
-      quiz_str.prepend('"quiz_scores":{')
-      quiz_str += ','
-      quiz_str = quiz_str.gsub('"N/A"', '20.0')
-      req.body.prepend(quiz_str)
+      req.body = add_quiz_scores(req.body)
```
Comment (krishnasaurabh) [inline app/controllers/reputation_web_service_controller.rb]: same here, can you check if this works as there is no return in add_quiz_scores

File: app/controllers/reputation_web_service_controller.rb [hunk 3]
```diff
@@ -163,87 +250,32 @@ def send_post_request
...
     else
       @@additional_info = ''
```
Comment (krishnasaurabh) [inline app/controllers/reputation_web_service_controller.rb:261]: can you create a separate method to set additional_info in a method in a dry fashion? also we might need to eliminate the use of class variables here, as it is not good to pass information around using class variables. we need to use function parameters to pass info, we can do that in the next PullRequest though, no need to change anything here.

File: app/controllers/reputation_web_service_controller.rb [hunk 4]
```diff
@@ -54,7 +54,7 @@ def get_peer_reviews_for_responses(reviewer_id, team_id, valid_response)
       peer_review_grades_list << [reviewer_id, team_id, review_grade]
     end
     peer_review_grades_list
-  end
+   end
```
Comment (krishnasaurabh) [inline app/controllers/reputation_web_service_controller.rb:57]: is this intended?

**Review by krishnasaurabh — CHANGES_REQUESTED** [review]
some methods have some assignments into req.body, but I see no returns for most of the methods, can you check it.

File: app/controllers/reputation_web_service_controller.rb (hunk 1 above)
Comment (gokul-koganti) [reply app/controllers/reputation_web_service_controller.rb]: Last processed object is automatically returned in ruby and the entities involved are passed by their references. So, it is not required to explicitly return it.

File: app/controllers/reputation_web_service_controller.rb (hunk 3 above)
Comment (gokul-koganti) [reply app/controllers/reputation_web_service_controller.rb:261]: Will deal with it in the next pr

File: app/controllers/reputation_web_service_controller.rb (hunk 4 above)
Comment (gokul-koganti) [reply app/controllers/reputation_web_service_controller.rb:57]: It was unintentional

**Review by krishnasaurabh — APPROVED** [review]
LGTM

**Pull request merged**